cypress-io/cypress
# fix(webpack-dev-server): Pass in jsconfig to get webpack config for Next.js >= 13.2.1

- Closes #25951

### Additional details


With the release of Next.js 13.2.1, we need to pass the JS/TS config into `getBaseWebpackConfig`, otherwise the config won't be respected when running the dev server. This PR adds some logic to get the user's JS/TS configuration and pass it to `getBaseWebpackConfig`.

### Steps to test


1. Create a Next.js project (v13.2.1) that uses the `src/` dir (it's in the generator options)
2. Configure Cypress for CT
3. Have a tsconfig.json file in the project root with these options
```js
{
  "compilerOptions": {
    // ...
    "paths": {
      "*": ["src/*"]
    },
  }
  "baseUrl": '.'
}
```
4. In `cypress/support/component.ts`, import a stylesheet like this
```js
import 'styles/globals.css'
```
5. Run a component test and verify that Webpack is able to resolve this import

Also test on < v13.2.1 and make sure that everything still works. 

### How has the user experience changed?


### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
